azurerm/internal/services/compute/ssh_public_key_data_source.go

```diff
@@ -67,7 +67,11 @@ func dataSourceSshPublicKeyRead(d *pluginsdk.ResourceData, meta interface{}) err
 	d.Set("resource_group_name", resGroup)
 
 	if props := resp.SSHPublicKeyResourceProperties; props != nil {
-		d.Set("public_key", props.PublicKey)
+		sshKey, err := utils.NormalizeSSHKey(*props.PublicKey)
+		if err != nil {
+			return fmt.Errorf("unable to remove invalid formatting and line breaks from SSH Public Key %q (Resource Group %q)", name, resGroup)
+		}
+		d.Set("public_key", sshKey)
```

Comment:
there's a crash here if this is nil, can we make this:

```suggestion
		publicKey := ""
		if props.PublicKey != nil {
			publicKey, err = utils.NormalizeSSHKey(*props.PublicKey)
			if err != nil {
				return fmt.Errorf("normalising public key: %+v", err)
			}
		}
		d.Set("public_key", publicKey)
```

Reply:
we return a pointer from these functions since if error is non-nil the object returned should be (to avoid coding errors)

instead we can make `publicKey` a `*string` via: `var publicKey *string` to avoid that

Reply:
ok, thanks.